src/main/java/org/opensearch/ad/rest/handler/AbstractAnomalyDetectorActionHandler.java

```diff
@@ -215,21 +217,42 @@ public AbstractAnomalyDetectorActionHandler(
 
     /**
      * Start function to process create/update/validate anomaly detector request.
-     * Check if anomaly detector index exist first, if not, will create first.
-     * 
-     * @throws IOException IOException from {@link AnomalyDetectionIndices#initAnomalyDetectorIndexIfAbsent(ActionListener)}
+     * If detector is not using custom result index, check if anomaly detector
+     * index exist first, if not, will create first. Otherwise, check if custom
+     * result index exists or not. If exists, will check if index mapping matches
+     * AD result index mapping and if user has correct permission to write index.
+     * If doesn't exist, will create custom result index with AD result index
+     * mapping.
      */
-    public void start() throws IOException {
-        if (!anomalyDetectionIndices.doesAnomalyDetectorIndexExist() && !this.isDryRun) {
-            logger.info("AnomalyDetector Indices do not exist");
-            anomalyDetectionIndices
-                .initAnomalyDetectorIndex(
-                    ActionListener.wrap(response -> onCreateMappingsResponse(response, false), exception -> listener.onFailure(exception))
-                );
-        } else {
-            logger.info("AnomalyDetector Indices do exist, calling prepareAnomalyDetectorIndexing");
-            logger.info("DryRun variable " + this.isDryRun);
-            validateDetectorName(this.isDryRun);
+    public void start() {
+        String resultIndex = anomalyDetector.getResultIndex();
+        // use default detector result index which is system index
+        if (resultIndex == null) {
+            createOrUpdateDetector();
+            return;
+        }
+
+        // user custom result index
+        anomalyDetectionIndices.initCustomResultIndexAndExecute(resultIndex, () -> createOrUpdateDetector(), listener);
```

Comment:
Before I only called on anomalyDetectionIndices.initAnomalyDetectorIndex if this.isDryRun was false which basically means the call was either Create/update API and not validate. Do we want to still call this even when just validating, basically it could create the index even before creation itself and just when validating.

Reply:
Oh, just saw your comments after merging. Currently we have no easy way to validate if user has permission to create index without trying to create index first. So I think it's ok for now as we have no other option. We can add this to AD public doc to let user know this. 
One workaround is to delete the custom result index after validating, but it's possible that another user running another detector on that index, so delete the result index is not safe.

Reply:
Another option is we can skip checking create permission for custom result index if we have concern about creating too many indices